pkg/system/phase2_creating.go

```diff
@@ -168,13 +171,49 @@ func (r *Reconciler) SetDesiredCoreApp() {
 	}
 }
 
-// ReconcileCredentialsRequest creates a CredentialsRequest resource if neccesary and returns
+// ReconcileBackinStoreCredentials creates a CredentialsRequest resource if neccesary and returns
 // the bucket name allowed for the credentials. nil is returned if cloud credentials are not supported
-func (r *Reconciler) ReconcileCredentialsRequest() error {
-	if !util.IsAWSPlatform() {
-		r.Logger.Info("not running in AWS. skipping ReconcileCredentialsRequest")
-		return nil
+func (r *Reconciler) ReconcileBackinStoreCredentials() error {
+	if util.IsAWSPlatform() {
+		return r.ReconcileAWSCredentials()
+	}
+	return r.ReconcileCephObjectstoreCredentials()
+
+}
+
+// ReconcileCephObjectstoreCredentials creates a ceph objectstore user if a ceph objectstore exists in the same namespace
+func (r *Reconciler) ReconcileCephObjectstoreCredentials() error {
+	r.Logger.Info("Not running in AWS. will attempt to create a ceph objectstore user")
+	util.KubeCheck(r.CephObjectstoreUser)
+	// create user if not already exists
+	if r.CephObjectstoreUser.UID == "" {
+		// list ceph objectstores and pick the first one
+		cephObjectStoresList := &cephv1.CephObjectStoreList{}
+		if !util.KubeList(cephObjectStoresList, &client.ListOptions{Namespace: options.Namespace}) {
+			return fmt.Errorf("failed to list ceph objectstores")
```

Comment:
Is this right to error here? What if we install noobaa without ceph? 
We should make sure that path works well too.

Reply:
you're right. shouldn't error here